### ISSUE TITLE
Bug formatting typescript typecast when in parentheses and using typescript parser

### ISSUE THREAD

**Issue by zxbodya** [prettier/prettier#11018]
**Prettier 2.3.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAbOMAEEBucBO+AlgCZyYC8mwAvgNwA6UARpZgBS4HFmYCGAZ35QAngEpGUEABoQEAA4wi0AclB9CEAO4AFDQlUo+qLXxGrZzfHzABrDAGV5NolADmyGPgCucWXABbZjgSMhIAGT53bz43OAAxCHwAvhgld2QQPm8YCBkQAAsYANQAdQKieAFnMDgHA0qiHEqRTLABCxBXAQIYHWs3FOQAM2Me2QArAQAPACFrO0c+ALhw1zgRsb8QKemHVzd0AEVvCHhN1HGQZ3we-EyYEXk4ATBiRXz5YlhS0hgC5AADgADLIvhAeqVrPJMl8XgQ8PkAI6neD9BSGLICAC0UDgIRC+XwcBRRGJ-ViQyQo0u2x6ASInh8dIOx1RG2pW1kMD4zF+JH+yAATNzrERUAcAMIQAJUkAvACs+W8PQAKrzDDSrjhfABJKBkWAON5ERQAQQNDke6AuPRoNCAA)

```sh
--parser typescript
```

**Input:**

```ts
let override = {};
b = (override as any);

```

**Output:**

```tsx
SyntaxError: ',' expected. (2:18)
  1 | let override = {};
> 2 | b = (override as any);
    |                  ^
  3 |
```

**Expected behavior:**
Should not have `SyntaxError` and to remove parentheses when formatting.

**Comment by zxbodya** [comment]
it looks `override` variable name is the reason of it - [playground link using different name](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAbOMAEFMF5PAC+A3ADpQBGemAFDgIYDOm9UAngJRlQgA0IEAA4wAltEbJQ9AE7SIAdwAKMhBJT1U8+mwn8K0+mADWGAMqDDIqAHNkMaQFc4-OAFsKcACaevAGVbWDvTWcABiENKu9DCiNsgg9A4wEHwgABYwrqgA6mki8IwWYHCmqvkiAG75bPFgjLogVoxw0jCKBtZRyABmGs38AFaMAB4AQgbGZvSucL5WcD19ziBDw6ZW1ugAig4Q8Iuo-SAW0s3S8TBsgnCMYNIiwqmC97DZIp4wacgAHAAM-M8IM1sgZBPFnjcWhUFvwAI67eDtIRqBKMAC0UDgXi8qWkcHhIjx7WCXSQvUOy2arhEdkclI22wRCzJS34MHoFDeHy+SAATGyDCJUBsAMIQVykkA3ACsqQczQAKhy1OSjhUnABJKA+WCmO4PGAAQW1pku6AOzUIhCAA)

**Comment by sosukesuzuki** [comment]
TypeScript cannot parse this https://www.typescriptlang.org/play?#code/MYewdgzgLgBCBuBTATsglgE0TAvDA3gL4DcAUKJLAEa4wAUCK6WMAhhG2AJ4CUZQA

**Comment by HolgerJeromin** [comment]
Is this a duplicate of my code problem?
Both compile in Typescript 4.7.2, but not in 4.6.4 (checked via TS Playground).
But both do throw `SyntaxError`s in prettier 2.6.2
I am happy to open a new issue here in this repo if they are really different...

**Prettier 2.6.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEcAeAHCAnGACMAGwEMBnUgwi4AHSjwYdJmJgEsw8AzKAHgBU8AXjzEoATwB8ACi4QISPPwCUiwbXqMt2ODACu2enIgBuOloC+ZqBbqFd3eYoAKxbMQC2uuNlK8Y4hhwEFyUpAB0PLwARvL2YpKSeAA+eHpQACZwXGxQcBnWIAA0IBAY7NCkyKBu2BAA7q46UFUoxIT1xOJVJdHuYADWugDKnnAAMrlwyFztpHC9-UMwwxjEYLkA5sgw2HoLqB7R+VkZ42KbesSbcABiOB6s7FDbbXowEMUgABYwHoQAdW+bHgpDWYDgwwQpBBbAAbiDxMhwOQvrl5rhnO5No8ZnMDgArUhoYZbewART0EHgeKoBzWvh8yICQVIYGwbHKXwwHNgALYGRg32QAA4AAwlHkQeYA9wYZE8uAYuHTEoARyp8CxZVaIDIAFo8vl8l8dBq2DosddcUhZnSSvMPGwdnsDjCXhTNdNbfiSixovzBcKkAAmP3uNiELYAYQgHhtqFIAFYvnp5vxiNFWnb5iU4fsAJKZBArdmcmAAQUywwC9lp8wsFiAA)

```sh
--parser typescript
```

**Input:**

```tsx
export class cls {
     static fn<T = any>(foo: T): T {
        return foo;
    };
}
let foo: Parameters<typeof cls.fn<boolean>> | undefined;

```

**Output:**

```tsx
SyntaxError: '>' expected. (6:34)
  4 |     };
  5 | }
> 6 | let foo: Parameters<typeof cls.fn<boolean>> | undefined;
    |                                  ^
  7 |
```

**Comment by HolgerJeromin** [comment]
Both problems are NOT fixed with prettier 2.7.0

**Comment by thorn0** [comment]
fixed in 2.7.1
